gen3-integration-tests/tests/test_gen3_workflows.py

```diff
@@ -0,0 +1,147 @@
+        """Unauthorized user should receive 403 when posting a S3 file."""
+        self.gen3_workflow.put_bucket_object_with_boto3(
+            content="dummy_S3_content",
+            object_path=f"{self.bucket_name}/{self.folder_name}/{self.file_name}",
+            user=self.invalid_user,
+            expected_status=403,
+        )
+
+    @pytest.mark.skipif(
+        not bucket_name,
+        reason="Skipping this test, since a valid s3 bucket name couldn't be found ",
+    )
+    def test_unauthorized_user_cannot_get_s3_file(self):
+        """Unauthorized user should receive 403 when trying to GET a S3 file."""
+        self.gen3_workflow.get_bucket_object_with_boto3(
+            object_path=f"{self.bucket_name}/{self.folder_name}/{self.file_name}",
+            user=self.invalid_user,
+            expected_status=403,
+        )
+
+    @pytest.mark.skipif(
+        not bucket_name,
+        reason="Skipping this test, since a valid s3 bucket name couldn't be found ",
+    )
+    def test_happy_path_upload_and_get_s3_file(self):
+        """
+        POST an S3 file in a `directoy/filename` format,
+        then GET with the `directory/` to test the list-object functionality
+        followed by GET `directory/filename`to verify the uploaded file exists.
+        """
+
+        input_content = "sample_S3_content"
+        self.gen3_workflow.put_bucket_object_with_boto3(
+            content=input_content,
+            object_path=f"{self.bucket_name}/{self.folder_name}/{self.file_name}",
+            user=self.valid_user,
+            expected_status=200,
+        )
+
+        # Fetch all the contents in the given folder to verify the list-objects functionality
+        response_contents = self.gen3_workflow.list_bucket_objects_with_boto3(
+            folder_path=f"{self.bucket_name}/{self.folder_name}/",
+            user=self.valid_user,
+            expected_status=200,
+        )
+        assert (
+            isinstance(response_contents, list) and len(response_contents) > 0
+        ), "Expected the function to return a list of at least one item"
+        assert (
+            "Key" in response_contents[0]
+            and response_contents["Key"] == f"{self.folder_name}/{self.file_name}"
+        ), "Expected folder integration-tests in the bucket to have file `test-input.txt`"
+
+        # Fetch the exact file to verify the get-object functionality
+        response_s3_object = self.gen3_workflow.get_bucket_object_with_boto3(
+            object_path=f"{self.bucket_name}/{self.folder_name}/{self.file_name}",
+            user=self.valid_user,
+            expected_status=200,
+        )
+
+        assert (
+            input_content in response_s3_object
+        ), "Stored and retrieved content should match"
+
+    @pytest.mark.skipif(
+        not bucket_name,
+        reason="Skipping this test, since a valid s3 bucket name couldn't be found ",
+    )
+    def test_happy_path_delete_s3_file(self):
```

Comment:
this test shouldn't depend on the previous one - either merge them into one, or repeat the PUT object logic